robot/client/client.go

```diff
@@ -991,10 +991,34 @@ func (rc *RobotClient) RestartModule(ctx context.Context, req robot.RestartModul
 	return nil
 }
 
-// Shutdown shuts down the robot. Will return "shutdown function not defined" error if
-// shutdown callback is not specified in robot options.
+// Shutdown shuts down the robot. May return DeadlineExceeded error if shutdown request times out,
+// or if robot server shuts down before having a chance to send a response. May return Unavailable error
+// if server is unavailable, or if robot server is in the process of shutting down when response is ready.
 func (rc *RobotClient) Shutdown(ctx context.Context) error {
 	reqPb := &pb.ShutdownRequest{}
 	_, err := rc.client.Shutdown(ctx, reqPb)
-	return err
+	if err != nil {
+		if status, ok := status.FromError(err); ok {
+			switch status.Code() {
+			case codes.Internal, codes.Unknown:
+				rc.Logger().CInfow(ctx, "robot shutdown successful")
+				return nil
+			case codes.Unavailable:
+				rc.Logger().CWarnw(ctx, "server unavailable, likely due to successful robot shutdown")
+				return err
+			case codes.DeadlineExceeded:
+				rc.Logger().CWarnw(ctx, "request timeout, robot shutdown may still be successful")
+				return err
+			case codes.OK, codes.Aborted, codes.AlreadyExists, codes.Canceled, codes.DataLoss,
+				codes.FailedPrecondition, codes.InvalidArgument, codes.NotFound, codes.PermissionDenied,
+				codes.ResourceExhausted, codes.OutOfRange, codes.Unimplemented, codes.Unauthenticated:
+				return err
+			default:
+				return err
+			}
```

Comment:
unless you meant to include special logic for these cases we should just collapse them into `default`

```suggestion
			default:
				return err
			}
```

Reply:
that's also what I would think lol, but the linter complains that the switch cases are not exhaustive and forces me to enumerate all of them. another option would be to just no-lint this section if you think that's better?

Reply:
oh that's surprising - I would have thought that `default` covers that. anyway, yes I think `no-lint` is clearer here.